src/ShareRoutes.test.jsx

```diff
@@ -1,13 +1,59 @@
-import React from 'react'
-import {render} from '@testing-library/react'
-import {mockRoutes} from './BaseRoutesMock.test'
-import ShareRoutes from './ShareRoutes'
-
-
-test('ShareRoutes', () => {
-  const {getByText} = render(mockRoutes(
-      <ShareRoutes
-        installPrefix={'/'}
-        appPrefix={'share'} /> ))
-  expect(getByText(/BLDRS/i)).toBeInTheDocument()
+import {
+  looksLikeLink,
+  trimToPath,
+  githubUrlOrPathToSharePath} from './ShareRoutes'
+
+
+const path = '/org/repo/branch/file.ifc'
+const pathAbc = '/org/repo/branch/a/b/c/file.ifc'
+const pathBlob = '/org/repo/blob/branch/file.ifc'
+const pathBlobAbc = '/org/repo/blob/branch/a/b/c/file.ifc'
+const tests = [
+  {s: 'http://www.github.com/org/repo/blob/branch/file.ifc', out: pathBlob},
+  {s: 'http://github.com/org/repo/blob/branch/file.ifc', out: pathBlob},
+  {s: 'http://www.github.com/org/repo/blob/branch/a/b/c/file.ifc', out: pathBlobAbc},
+  {s: 'http://github.com/org/repo/blob/branch/a/b/c/file.ifc', out: pathBlobAbc},
+  {s: 'https://www.github.com/org/repo/blob/branch/file.ifc', out: pathBlob},
+  {s: 'https://github.com/org/repo/blob/branch/file.ifc', out: pathBlob},
+  {s: 'github.com/org/repo/blob/branch/file.ifc', out: pathBlob},
+  {s: 'githubcom/org/repo/blob/branch/file.ifc', out: pathBlob},
+  {s: '/org/repo/blob/branch/file.ifc', out: pathBlob},
+  {s: '/org/repo/branch/file.ifc', out: path},
+  {s: '/org/repo/blob/branch/a/b/c/file.ifc', out: pathBlobAbc},
+  {s: '/org/repo/branch/a/b/c/file.ifc', out: pathAbc},
+  {s: 'www.google.com', err: 'Not a url with path: www.google.com'},
+]
+
+
+describe('ShareRoutes', () => {
+  test('looksLikeLink', () => {
+    tests.forEach((pair) => {
+      expect(looksLikeLink(pair.s)).toBe(pair.out !== undefined)
+    })
```

Comment:
who is this test cathing the invalid URLS?

Reply:
the function under test here just returns a boolean and doesn't throw exception.